src/realm/ucx/ucp_internal.cc

```diff
@@ -2381,8 +2423,16 @@ namespace UCP {
         goto err_update_pending;
       }
       *req = *req_prim;
-      CHKERR_JUMP(!worker->ep_get(target, remote_dev_index, &req->ucp.ep),
+
+      if (!worker->ep_get(target, remote_dev_index, &req->ucp.ep)) {
```

Comment:
Maybe we can move the check and add_remote_ep into `UCPMessageImpl::commit` instead of duplicating them in `commit_multicast` and `commit_unicast`.